File: framework/starter-logging/src/main/java/cn/bestwu/logging/FilesProperties.java
package cn.bestwu.logging;

import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * logging file 配置
 *
 * @author Peter Wu
 */
@ConfigurationProperties("logging.files")
public class FilesProperties {

  private String path;
  private String maxFileSize = "100MB";
  private Integer maxHistory = 100;
  /**
   * 是否启动时滚动日志
   */
  private boolean rolloverOnStart = false;
  /**
   * 记录所有日志于一个文件
   */
  private boolean logAll = true;

  public String getPath() {
    return path;
  }

  public void setPath(String path) {
    this.path = path;
  }

  public String getMaxFileSize() {
    return maxFileSize;
  }

  public void setMaxFileSize(String maxFileSize) {
    this.maxFileSize = maxFileSize;
  }

  public Integer getMaxHistory() {
    return maxHistory;
  }

  public void setMaxHistory(Integer maxHistory) {
    this.maxHistory = maxHistory;
  }

  public boolean isRolloverOnStart() {
    return rolloverOnStart;
  }

  public void setRolloverOnStart(boolean rolloverOnStart) {
    this.rolloverOnStart = rolloverOnStart;
  }

  public boolean isLogAll() {
    return logAll;
  }

  public void setLogAll(boolean logAll) {
    this.logAll = logAll;
  }
}

File: web/src/main/kotlin/top/bettercode/summer/logging/FilesProperties.kt
package top.bettercode.summer.logging

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * 配置 logging file
 *
 * @author Peter Wu
 */
@Suppress("ConfigurationProperties")
@ConfigurationProperties("summer.logging.files")
open class FilesProperties {
    /**
     * 查看路径
     */
    var viewPath: String? = null
    /**
     * 路径
     */
    var path: String? = null
    /**
     * 最大文件大小
     */
    var maxFileSize = "10MB"

    /**
     * 最大历史记录数量
     */
    var maxHistory = 100000

    /**
     * 启动时是否滚动日志
     */
    var isRolloverOnStart = true
}